### PR TITLE
Allow handles with 'copy' fate to persist to firebase

### PR DESCRIPTION
**Pull request by shans**
1) Ensure that cloning a handle into a firebase-collection persists.
2) re-add btoa/atob on firebase keys to prevent invalid keys being generated.

### CODE REVIEW

**Comment by shans** [comment]
Fixes #1699

File: runtime/storage/firebase-storage.js [hunk 1]
```diff
@@ -718,6 +728,15 @@ class FirebaseCollection extends FirebaseStorageProvider {
 
   async cloneFrom(handle) {
     this.fromLiteral(await handle.toLiteral());
+    // Don't notify about the contents that have just been cloned.
+    // However, do record local changes for persistence.
+    for (let item of this._model._items.values()) {
+      assert(item.value.id !== undefined);
+      console.log(item.value, item.keys);
```
Comment (shaper) [inline runtime/storage/firebase-storage.js]: Probably remove this?

File: runtime/storage/firebase-storage.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import {StorageProviderBase} from './storage-provider-base.js';
 import {firebase} from '../../platform/firebase-web.js';
 import {assert} from '../../platform/assert-web.js';
 import {KeyBase} from './key-base.js';
+import {atob} from '../../platform/atob-web.js';
```
Comment (shaper) [inline runtime/storage/firebase-storage.js]: Add a test that would have caught the absence of this?

File: runtime/storage/firebase-storage.js (hunk 1 above)
Comment (shans) [reply runtime/storage/firebase-storage.js]: Done.

File: runtime/storage/firebase-storage.js (hunk 2 above)
Comment (shans) [reply runtime/storage/firebase-storage.js]: There is one, now that the decodeKey function is actually used.

**Pull request merged**